src/dxtbx/format/FormatNXmxDLS.py

```diff
@@ -13,36 +12,63 @@
 from dxtbx.format.FormatNXmx import FormatNXmx
 
 
-def get_bit_depth_from_meta(meta_file_name):
+def get_bit_depth_from_meta(meta_file_name: Path) -> int:
     with h5py.File(meta_file_name) as f:
-        return int(f["/_dectris/bit_depth_image"][()])
+        with contextlib.suppress(KeyError):
+            return int(f["/_dectris/bit_depth_image"][()])
 
+        # This might be a very old meta file, that only had 'datatype'?
+        if "datatype" in f:
+            unique_vals = np.unique(f["datatype"])
+            if not len(unique_vals) == 1:
+                raise RuntimeError(
+                    f"Error: Could not determine bit depth from metafile {meta_file_name} (multiple bit depths)"
+                )
+            return np.dtype(unique_vals[0]).itemsize * 8
 
-def find_meta_filename(master_like: Union[str, Path]) -> Union[str, Path]:
+    raise RuntimeError("Cannot determine metafile bit depth")
+
+
+def find_meta_filename(master_like: Path) -> Path:
```

Comment:
Does this work with `foo.nxs` with no internal references to `foo_meta.h5`? I can see `foo_master.h5` etc. being easy

Reply:
Yes, I tested with David's test case, which was using the `.nxs`. I just also checked with `_master.h5` and that works too.